caikit/interfaces/common/data_model/remote.py

```diff
@@ -0,0 +1,139 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This file contains interfaces required to connect to Remote servers
+"""
+
+# Standard
+from dataclasses import field
+from http.client import HTTP_PORT, HTTPS_PORT
+from pathlib import Path
+from typing import Optional
+
+# First Party
+from py_to_proto.dataclass_to_proto import Annotated, Dict, FieldNumber
+import alog
+
+# Local
+from caikit.core.data_model import PACKAGE_COMMON, DataObjectBase, dataobject
+from caikit.core.exceptions import error_handler
+
+log = alog.use_channel("CNNCTDM")
+error = error_handler.get(log)
+
+
+@dataobject(PACKAGE_COMMON)
+class ConnectionTlsInfo(DataObjectBase):
+    """Helper dataclass to store information regarding TLS information."""
+
+    enabled: Annotated[bool, FieldNumber(1)] = False
```

Comment:
Wouldn't this be implied by the truthiness of `ca_file`? I suppose for HTTP with "standard certs" you could enable it but not not set an explicit `ca_file`

Reply:
You're spot on! This is required for `HTTP` with insecure_verify to determine if TLS should be used